### DISCUSSION TITLE
p-convergence

### DISCUSSION THREAD

**Discussion by fdrmrc** [fdrmrc/non_matching_test_suite#37] [Show and tell]
After the meeting I've been investigating the convergence in $p$ against $\sqrt{DoFs}$ for the different types of agglomerated grids for $\Omega = [0,1]^2$:
1. structured square
2. unstructured square (gmsh)

with manufactured solution $u=\sin(\pi x)\sin(\pi y)$.

- - - - - - - - - - - - -
## 1. Structured square

The fixed mesh is with 1024 agglomerates. The Rtree version is not shown as it's "perfectly" structured, I'm displaying here only the METIS' one. The underlying triangulation is quite fine so the polygons are not bad for this mesh.

![immagine](https://github.com/fdrmrc/Polydeal/assets/50499769/0295ab14-01f8-42a1-9348-48556b87fa24)

Here's the p-convergence for $p=1,2,3,4$.
![p_convergence_structured](https://github.com/fdrmrc/Polydeal/assets/50499769/f91a5096-ea66-4eca-9497-6edd4b8e59e7)

## 2. Unstructured square

The fixed meshes are the following ones, with 364 agglomerates.
Rtree | METIS  
:-------------------------:|:-------------------------:
![Rtree](https://github.com/fdrmrc/Polydeal/assets/50499769/a0b42521-6d61-4d89-8988-f4f6eec8a463) |  ![METIS](https://github.com/fdrmrc/Polydeal/assets/50499769/63ea2cce-36bd-4f58-9753-cc2014730500)

Here's the p-convergence for $p=1,2,3,4$.

![p_convergence_unstructured](https://github.com/fdrmrc/Polydeal/assets/50499769/f8ad5ea3-e5ce-40b0-b814-49ef3feb13e9)

- - - - - - - - - -

@andreacangiani @luca-heltai

**Comment by andreacangiani** [comment]
Perfect. We see indeed that in this case the gain with r-tree is superior in the structured case. In the unstructured case the two fixed meshes are similar.

**Reply by fdrmrc** [reply]
Yes. If you decrease the number of agglomerates, the difference is more prominent also in the unstructured case. Here the unstructured test with 91 agglomerates (instead of 364 as before):

$p=1,2,3,4,5$.
![p_convergence_unstructured_91](https://github.com/fdrmrc/Polydeal/assets/50499769/ab642997-23ba-4396-bc43-5cd4c3475a23)

**Comment by fdrmrc** [comment]
@luca-heltai @andreacangiani 

I've validated the code with some 3D sanity checks, and a manufactured solution $u=sin(\pi x) \sin(\pi y) \sin(\pi z)$. 

- $p$ convergence ($p=1,2,3,4$), $\Omega=[0,1]^3$ with a _fixed_ agglomerated mesh of 100 polytopes:
![p_convergence_cube](https://github.com/fdrmrc/Polydeal/assets/50499769/17a9d209-0b82-4229-a526-0aa48da7f966)
- $p$ convergence ($p=1,2,3,4$), $\Omega = \mathcal{B}_1(0)$ with a _fixed_ agglomerated mesh of $100$ polytopes:
![p_convergence_ball](https://github.com/fdrmrc/Polydeal/assets/50499769/eb381961-8201-4a73-8094-c1811ca0b6f9)
